### ISSUE TITLE
SA1134 fires erroneously on lambda expression with attribute

### ISSUE THREAD

**Issue by alexrp** [DotNetAnalyzers/StyleCopAnalyzers#3623]
![image](https://user-images.githubusercontent.com/44076/229246220-eda1c7a6-17ec-41bf-a114-6674564fa99c.png)
![image](https://user-images.githubusercontent.com/44076/229246229-def8dff2-0685-4e23-8f0f-73c630830587.png)

**Comment by alexrp** [comment]
Just to expand a bit: Putting an attribute on the same line as (e.g.) a regular method declaration does not seem to trigger this rule:

```csharp
[Pure] int Foo() => 42;
```

So it seems like the same ought to be allowed for a single attribute on a lambda expression.

**Comment by sharwell** [comment]
> Just to expand a bit: Putting an attribute on the same line as (e.g.) a regular method declaration does not seem to trigger this rule:
> 
> ```cs
> [Pure] int Foo() => 42;
> ```
> 
> So it seems like the same ought to be allowed for a single attribute on a lambda expression.

That seems like an oversight. We have a test validating something very similar:

https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/ef1f91c75c7fd4ea7f7e2872ea38f8e4c4fe8ccc/StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1134UnitTests.cs#L345-L348

I believe the current behavior of SA1134 is correct for the lambdas, but needs to be updated for methods.

**Comment by alexrp** [comment]
Actually, you can disregard my comment above. That snippet *does* trigger SA1134, I just missed it.

---

That aside, it seems undesirable to force people to rewrite a one-liner like:

```csharp
SomeMethod([SomeAttribute] x => Work(x));
```

To:

```csharp
SomeMethod(
    [SomeAttribute]
    x => Work(x));
```

Aesthetically, I guess I would consider this case somewhat similar to putting attributes on parameters. There, I would try to fit them onto the same line when possible and only resort to multiple lines if I (somehow) had a monstrosity with like 5+ attributes on a parameter.

**Comment by sharwell** [comment]
How often does that come up in practice? I've never seen it myself, but that might just be a characteristic of the code I've been working in.

**Comment by alexrp** [comment]
For me, almost never. I think I've only ever reached for attributes on lambdas when I wanted to apply `[SuppressMessage]`, and in those cases it's usually a multi-line lambda anyway.

[But I recall that the attributes-on-lambdas feature was motivated by ASP.NET Core needing it.](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-10.0/lambda-improvements#motivation) So maybe it's more common in that world? 🤷

**Comment by bjornhellander** [comment]
I get the warning on the code you provided in the last comment and I can't find any exceptions in the analyzer code. Can you provide a link to a complete example where you don't get the warning?

**Comment by alexrp** [comment]
This has actually started coming up more often than I would have expected: While annotating some of my projects for trimming, I find myself putting `[UnconditionalSuppressMessage(...)]` on lambdas fairly frequently.

**Comment by alexrp** [comment]
Also worth noting that even if you do what this rule wants you to do, you then hit `IDE0055`:

![image](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/assets/44076/8c160c34-b798-42b7-9590-f99bdddf468e)

This leaves you with no good options AFAICT.
